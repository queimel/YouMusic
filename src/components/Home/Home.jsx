import React, { Component } from 'react';
import Search from '../search/search'
import SearchList from '../searchList/SearchList'


class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
            songItems : []
        }


    }
    componentDidMount(){
        this.setState({playerState: this.props.showMiniPlayer})

        if(this.state.songItems === null){
            this.setState({list: this.lastSearches() })
        }

        console.log(this.state.songItems)

    }

    lastSearches(){
        return JSON.parse(localStorage.getItem("latestSearches"));
    }

    handleApiQuery(e) {
        let terms = e.target.value;
        const ApiKey = 'AIzaSyAUd28jjI7Xv_jB4wypxv-BiUPd6GHBeiA'

        let endpoint = `https://www.googleapis.com/youtube/v3/search?key=${ApiKey}&q=${terms}&part=snippet,id&order=viewCount&maxResults=10&type=video`
        
        fetch(endpoint)
            .then((response) => {
                return response.json()
            })
            .then((videos) => {
                const { items } = videos
                let videosArray =  items.map(function(item){
                    return {
                        "songId": item.id.videoId,
                        "songTitle": item.snippet.title,
                        "songImg": item.snippet.thumbnails.medium.url                        
                    }
                })

                this.setState({ songItems: videosArray })
            })
    }

    setVideoSelected(videoId){
     
        let song = this.state.songItems[videoId]
        console.log(song)
        this.props.songInfo(song)
        
    }
    
    render() {

        return (
            <div id="home">
                <header>
                    <Search onSearch ={ this.handleApiQuery.bind(this)}/>
                </header>
                <section id="results">
                    <SearchList 
                    results = {this.state.songItems} 
                    onClickedVideo={this.setVideoSelected.bind(this)}
                />
                </section>
            </div>
        );
    }
}

export default Home;