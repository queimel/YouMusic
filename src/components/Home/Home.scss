h3{
    color: white;
    border-bottom: #333 solid 1px;
    font-size: 0.9em;
    font-weight: 300;
    padding-bottom: 1em;
}