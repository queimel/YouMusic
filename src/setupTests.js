import requestAnimationFrame from './tempPolyfills';
 
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = require('jest-fetch-mock')
window.fetch = require('jest-fetch-mock')
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

