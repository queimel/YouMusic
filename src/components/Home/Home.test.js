import React from 'react';
import { shallow } from 'enzyme';
import setupTests from '../../setupTests';
import tempPolyfills from '../../tempPolyfills';
import Home from './Home';



const home = shallow(<Home />);

it('renders correctly', () => {
    expect(home).toMatchSnapshot();
});

describe('API Caller', () => {
    beforeEach(() => {
    
    });

    

    it('should fetch videos', () => {
    
        return home.instance().getVideos('nouvelle gaia')
            .then((videos) => {
                const { items } = videos
                expect(items[0].snippet.title).toEqual('Nouvelle Gaia - Get Free From Fear');
            });

            
    });
});