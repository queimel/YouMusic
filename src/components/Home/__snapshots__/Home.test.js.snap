// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <Search
            onSearch={[Function]}
          />
        </header>,
        <section
          id="results"
        >
          <SearchList
            onClickedVideo={[Function]}
            results={Array []}
          />
        </section>,
      ],
      "id": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Search
            onSearch={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSearch": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <SearchList
              onClickedVideo={[Function]}
              results={Array []}
            />,
          ],
          "id": "results",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClickedVideo": [Function],
              "results": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <Search
              onSearch={[Function]}
            />
          </header>,
          <section
            id="results"
          >
            <SearchList
              onClickedVideo={[Function]}
              results={Array []}
            />
          </section>,
        ],
        "id": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Search
              onSearch={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onSearch": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <SearchList
                onClickedVideo={[Function]}
                results={Array []}
              />,
            ],
            "id": "results",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClickedVideo": [Function],
                "results": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
